### PR TITLE
fix(compilation): preserve COW parameter views and FP64 training

### PR DESCRIPTION
**Pull request by ooples**
## Summary
- prepare copy-on-write training parameters before graph tracing and reject unsafe plain training scopes
- preserve writable-view isolation across COW peers while adding eager and compiled gradients for metadata, slice, and narrow views
- bind FP64 compiled kernels only to verified live backing arrays, materialize deferred CPU gradients before optimization, and preserve exact GELU semantics
- canonicalize tied parameters so each identity is updated once while accepting original registration group maps and rejecting conflicts
- support contiguous offset and non-contiguous parameter views in grouped/ungrouped FP32 and FP64 fused optimizers
- use zero-copy backing+offset binding for contiguous storage and reusable gather/scatter staging for strided/native storage
- keep dense SIMD and multi-tensor fast paths while routing unsupported view layouts through stride-aware compiled forward/backward fallbacks
- make gradient clipping view-aware and distinguish views from unmaterialized streaming weights
- suspend ambient graph recording while executing a compiled training step

## Why
AiDotNet PR ooples/AiDotNet#1789 exposed silent FP64 training corruption and clone divergence in several learnability shards. The defects crossed storage ownership, graph capture, pooled backing arrays, deferred gradients, parameter views, and tied parameters, so fixing only a model or disabling fusion would hide the underlying failures.

## Validation
- Release build: AiDotNet.Tensors.Tests, net471 - passed with 0 errors
- Release build: AiDotNet.Tensors.Tests, net10.0 - passed with 0 errors
- final focused COW/view/optimizer/FP64/GPU-mock/checkpointing regression set - 179 passed, 0 failed
- new parameter-view integration tests - 6 passed, covering offset sentinels, repeated strided staging, COW peers, MatMul fallback, grouped FP32/FP64, Adam multi-tensor fallback, and clipping
- prior CI's four stale COW-scope failures migrated to the safe GraphMode.EnableTraining(parameters) contract
- git diff --check - clean

The full test suite was not repeated locally; CI supplies the broader matrix while local validation stays focused on affected paths.

## Design note
Dense parameters remain zero-allocation and retain SIMD/multi-tensor fusion. Contiguous views carry explicit base offsets without copying. Non-contiguous or non-array CPU views use one persistent logical staging buffer per operand, refreshed and committed each step; GPU offset uploads likewise reuse scratch storage. Read-only, sparse, or unattached non-CPU storage fails fast. Specialized kernels bind only verified zero-offset live arrays, while other layouts remain supported by stride-aware compiled fallbacks.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved training compilation and replay reliability, including safer handling of shared parameters and tied parameters.
  * Fixed gradient propagation through reshape, transpose, slice, narrow, and sub-tensor views.
  * Improved optimizer updates and gradient clipping for offset, non-contiguous, and pooled tensor storage.
  * Corrected CUDA graph replay boundaries and deferred gradient materialization.
  * Improved FP64 and GELU training behavior, including reliable fallback handling for non-contiguous data.

* **Tests**
  * Added comprehensive coverage for copy-on-write isolation, optimizer parameter views, live tensor storage, gradient propagation, and pooled outputs.

### CODE REVIEW

File: src/AiDotNet.Tensors/LinearAlgebra/Tensor.cs [hunk 1]
```diff
@@ -156,6 +156,11 @@ internal Tensor(Vector<T> data, int[] shape, int[] strides, int storageOffset, T
     /// </summary>
     private Tensor<T> CreateStorageView(int[] shape, int[] strides, int storageOffset)
     {
+        // Exposing a retained alias is semantically a write-capability escape:
+        // future writes through either tensor bypass the other's COW gate. Make
+        // this tensor private first so the new view can never remain attached
+        // to an unrelated COW peer after either peer detaches.
+        EnsureOwnedForWrite();
```
[the remark this answers is not in the document]
Comment (ooples) [reply src/AiDotNet.Tensors/LinearAlgebra/Tensor.cs]: Addressed in 6fb3621 with an alias-family COW design. Metadata views remain O(1) and join their creator family; the first write atomically detaches the entire source/view family while preserving isolation from the independent clone family. View creation and family registration are serialized against concurrent detach so `_data`/`_storage` cannot be captured inconsistently. Coverage includes view-first mutation, source-first mutation, view chains, and a concurrent view-creation/first-write stress test on net10 and net471.
